cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityTopicOperator.java

```diff
@@ -264,6 +305,41 @@ public Secret generateSecret(ClusterCa clusterCa, boolean isMaintenanceTimeWindo
             CERT_SECRET_KEY_NAME, labels, ownerReference, isMaintenanceTimeWindowsSatisfied);
     }
 
+    /**
+     * Creates the Secret containing Cruise Control API auth credentials.
+     * 
+     * @param oldSecret The old secret.
+     *                           
+     * @return The generated Secret.
+     */
+    public Secret generateCruiseControlApiSecret(Secret oldSecret) {
+        return ModelUtils.createSecret(KafkaResources.entityTopicOperatorCcApiSecretName(cluster), namespace, labels, ownerReference, 
+            generateCruiseControlApiCredentials(oldSecret), Collections.emptyMap(), Collections.emptyMap());
+    }
+    
+    private static Map<String, String> generateCruiseControlApiCredentials(Secret oldSecret) {
+        if (oldSecret != null) {
+            // The credentials should not change with every release
+            // So if the secret with credentials already exists, we re-use the values
+            // But we use the new secret to update labels etc. if needed
+            var data = oldSecret.getData();
+            var username = data.get(API_TO_ADMIN_NAME_KEY);
+            var password = data.get(API_TO_ADMIN_PASSWORD_KEY);
+            if (username == null || username.isBlank() || password == null || password.isBlank()) {
+                throw new RuntimeException(format("Secret %s is invalid", oldSecret.getMetadata().getName()));
+            } else {
+                return data;
+            }
+        } else {
+            PasswordGenerator passwordGenerator = new PasswordGenerator(16);
+            String apiToAdminPassword = passwordGenerator.generate();
+            return Map.of(
+                API_TO_ADMIN_NAME_KEY, Util.encodeToBase64(API_TO_ADMIN_NAME),
+                API_TO_ADMIN_PASSWORD_KEY, Util.encodeToBase64(apiToAdminPassword)
+            );
+        }
+    }
```

Comment:
Should method be consistent with what we have in the `generateCruiseControlApiCredentials`  method of the Cruise Control class and check + generate API auth file too?

Reply:
There is no need, as this content is added to the CC's API auth file, which is the source of truth for authentication.